File: eemrs-server-master/src/main/java/com/liu/eemrsserver/domain/LabReport.java
package com.liu.eemrsserver.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigInteger;
import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class LabReport {
    private Long id;
    private String patientIdHashCode;
    private String reportToken;
    private String reportPayloadCipher;
    private String departmentCipher;
    private BigInteger reportTimeOpe;
    private String reportTypeCipher;
    private String imageCipherUrl;
    private Date createdAt;
    private Date updatedAt;

    private String reportPayload;
    private String department;
    private BigInteger reportTime;
    private String reportType;
}

File: frontend-vue/src/api/labReport.ts
import request from './request'

export interface LabReportQuery {
  department: string
  queryTime: string
  patientIdNumber?: string
}

export interface LabReport {
  id?: number
  patientIdHashCode?: string
  reportToken?: string
  reportPayloadCipher?: string
  departmentCipher?: string
  reportTimeOpe?: string
  reportTypeCipher?: string
  imageCipherUrl?: string
  reportPayload?: string
  department?: string
  reportTime?: string
  reportType?: string
}

export interface LabReportSearchResponse {
  latestReport: LabReport | null
  historyReports: LabReport[]
}

export function queryLabReportsByDepartmentTime(params: LabReportQuery) {
  return request.get<never, LabReportSearchResponse>('/lab-reports/search-by-dept-time', { params })
}
